<launch>
    <node pkg="audio_utils" exec="capture_node" name="capture_node">
        <param name="backend" value="alsa"/>
        <param name="device" value="default"/>
        <param name="format" value="signed_16"/>
        <param name="channel_count" value="1"/>
        <param name="sampling_frequency" value="16000"/>
        <param name="frame_sample_count" value="1024"/>
        <param name="merge" value="false"/>
        <param name="gain" value="1.0"/>

        <remap from="audio_out" to="beat_detector_audio_in"/>
    </node>

    <node pkg="audio_utils" exec="beat_detector_node" name="beat_detector_node">
        <param name="sampling_frequency" value="16000"/>
        <param name="frame_sample_count" value="64"/>
        <param name="oss_fft_window_size" value="1024"/>
        <param name="flux_hamming_size" value="15"/>
        <param name="oss_bpm_window_size" value="1024"/>
        <param name="min_bpm" value="50.0"/>
        <param name="max_bpm" value="180.0"/>
        <param name="bpm_candidate_count" value="10"/>

        <remap from="audio_in" to="beat_detector_audio_in"/>
        <remap from="bpm" to="dance/bpm"/>
        <remap from="beat" to="dance/beat"/>
    </node>

    <node pkg="dance" exec="head_dance_node.py" name="head_dance_node">
        <param name="movement_file" value="$(find-pkg-share dance)/movements/head_movements.json"/>

        <remap from="pose/filter_state" to="head_dance/filter_state"/>
        <remap from="bpm" to="dance/bpm"/>
        <remap from="beat" to="dance/beat"/>
    </node>

    <node pkg="dance" exec="torso_dance_node.py" name="torso_dance_node">
        <param name="movement_file" value="$(find-pkg-share dance)/movements/torso_movements.json"/>

        <remap from="pose/filter_state" to="torso_dance/filter_state"/>
        <remap from="bpm" to="dance/bpm"/>
        <remap from="beat" to="dance/beat"/>
    </node>

    <node pkg="dance" exec="led_dance_node.py" name="led_dance_node">
        <param name="led_colors_file" value="$(find-pkg-share dance)/movements/led_colors.json"/>

        <remap from="set_led_colors/filter_state" to="led_dance/filter_state"/>
        <remap from="bpm" to="dance/bpm"/>
        <remap from="beat" to="dance/beat"/>
    </node>
</launch>
